fdbserver/GrvProxyServer.actor.cpp

```diff
@@ -361,16 +366,28 @@ ACTOR Future<Void> queueGetReadVersionRequests(Reference<AsyncVar<ServerDBInfo>>
 	loop choose {
 		when(GetReadVersionRequest req = waitNext(readVersionRequests)) {
 			// WARNING: this code is run at a high priority, so it needs to do as little work as possible
-			if (stats->txnRequestIn.getValue() - stats->txnRequestOut.getValue() >
-			    SERVER_KNOBS->START_TRANSACTION_MAX_QUEUE_SIZE) {
-				++stats->txnRequestErrors;
+			if (req.priority >= TransactionPriority::DEFAULT &&
+			    stats->systemAndDefaultTxnRequestIn.getValue() - stats->systemAndDefaultTxnRequestOut.getValue() >
+			        SERVER_KNOBS->START_SYSTEM_AND_DEFAULT_TRANSACTION_MAX_QUEUE_SIZE) {
+				++stats->systemAndDefaultTxnRequestErrors;
 				// FIXME: send an error instead of giving an unreadable version when the client can support the error:
 				// req.reply.sendError(proxy_memory_limit_exceeded());
 				GetReadVersionReply rep;
 				rep.version = 1;
 				rep.locked = true;
 				req.reply.send(rep);
 				TraceEvent(SevWarnAlways, "ProxyGRVThresholdExceeded").suppressFor(60);
+			} else if (req.priority < TransactionPriority::DEFAULT &&
+			           stats->batchTxnRequestIn.getValue() - stats->batchTxnRequestOut.getValue() >
+			               SERVER_KNOBS->START_BATCH_TRANSACTION_MAX_QUEUE_SIZE) {
+				++stats->batchTxnRequestErrors;
+				// FIXME: send an error instead of giving an unreadable version when the client can support the error:
+				// req.reply.sendError(proxy_memory_limit_exceeded());
```

Comment:
This is a bit of a tangent, but we should do this FIXME. This code wasn't meant to have lived on past release-6.3.